dot/parachain/types/types.go

```diff
@@ -740,6 +747,54 @@ type Subsystem interface {
 	Stop()
 }
 
+type Present struct{}
+
+// UpgradeRestriction a possible restriction that prevents a parachain
+// from performing an upgrade
+// TODO: should be scale encoded/decoded
```

Comment:
If it has already been addressed, can we remove this to-do comment?
Do we need this type in multiple subsystems?

Reply:
the type `UpgradeRestriction` and its variant `Present` is used to validate the candidate, this validation happens only in `prospective-parachains` subsystem, this is part of `Constraints` type and because of that it should be in the `types` pkg